dotcom-rendering/src/server/lib/logging-store.ts

```diff
@@ -39,6 +40,32 @@ export const recordTypeAndPlatform = (
 	}
 };
 
+export const recordPageId = (pageId: string): void => {
+	const { request } = loggingStore.getStore() ?? {};
+
+	if (request) {
+		request.pageId = pageId;
+	}
+};
+
+type RecordablePeriod = 'total' | 'json' | 'enhance' | 'render';
+
+export const recordTimeStart = (metric: RecordablePeriod): void => {
+	const { timing } = loggingStore.getStore() ?? {};
+
+	if (timing) {
+		timing[metric] = performance.now();
+	}
+};
+
+export const recordTimeStop = (metric: RecordablePeriod): void => {
+	const { timing } = loggingStore.getStore() ?? {};
+
+	if (timing) {
+		timing[metric] = timing[metric] ?? 0 - performance.now();
+	}
```

Comment:
Would this not return negative values? If a value already exists, it will be less than or equal to the new call to `performance.now()`